pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -108,6 +108,29 @@ class CollectionViewSet(
     serializer_class = CollectionSerializer
     pagination_class = LimitOffsetPagination
 
+    def filter_queryset(self, queryset):
+        """
+        Filter Repository related fields.
+        """
+        queryset = super().filter_queryset(queryset)
+
+        distro = get_object_or_404(AnsibleDistribution, base_path=self.kwargs["path"])
+
+        if self.request.query_params.get("deprecated", "").lower() in ["true", "yes", "1"]:
+            deprecated = True
+        else:
+            deprecated = False
+
+        repository = distro.repository.cast()
+        deprecated_pks = repository.collection_memberships.values_list("collection_id", flat=True)
+
+        if deprecated_pks and deprecated:
+            queryset = queryset.filter(collection_id__in=deprecated_pks)
+        elif deprecated_pks:
+            queryset = queryset.exclude(collection_id__in=deprecated_pks)
+
+        return queryset
+
```

Comment:
I'm worried about performance here. It looks like we have to basically pull all the deprecated collections and paste them into the query. Not sure how django handles this, but if we end up lots of deprecated collections, this seems like it could be a big drag on the database.